molsysmt/_private/digestion/box.py

```diff
@@ -1,76 +1,72 @@
-    elif len(shape)==2:
-        if shape[1]==3:
-            output = box_lengths
-        else:
-            raise ValueError('box_lengths array with has not the correct shape.')
-    else:
-        raise ValueError('box_lengths array with has not the correct shape.')
-
-    return output
-
-def digest_box_lengths(box_lengths):
-
-    output = None
-    unit = puw.get_unit(box_lengths)
-    box_lengths_value = puw.get_value(box_lengths)
-    box_lengths_value = digest_box_lengths_value(box_lengths_value)
-    output = box_lengths_value*unit
 
-    return output
+def digest_box_vectors_value(box_vectors):
+    """ Checks if box_vectors has the correct shape.
 
-def digest_box_angles_value(box_angles):
+        The array should have shape (n, 3) where n is any integer.
+        However, if a list, tuple is passed it will be converted
+        to an array with the desired shape. Also, if an array of
+        shape (3, ) is passed its dimensions will be expanded to
+        (1, 3). If an array with rank > 2 is passed and exception
+        is raised.
 
-    output = None
+        Parameters
+        ----------
+        box_vectors : np.ndarray, list or tuple
+            A quantity with the box lengths.
 
-    if type(box_angles) is not np.ndarray:
-        box_angles = np.array(box_angles)
+        Raises
+        ------
+        IncorrectShapeError
+            If box_vectors doesn't have the correct shape.
+    """
+    if not(isinstance(box_vectors, np.ndarray)):
+        box_vectors = np.array(box_vectors)
 
-    shape = box_angles.shape
+    shape = box_vectors.shape
 
-    if len(shape)==1:
-        if shape[0]==3:
-            output = np.expand_dims(box_angles, axis=0)
+    if len(shape) == 1:
+        if shape[0] == 3:
+            box_vectors = np.expand_dims(box_vectors, axis=0)
         else:
-            raise ValueError('box_angles array with has not the correct shape.')
-    elif len(shape)==2:
-        if shape[1]==3:
-            output = box_angles
-        else:
-            raise ValueError('box_angles array with has not the correct shape.')
+            raise IncorrectShapeError(expected_shape="(1, 3)", actual_shape=str(shape))
+    elif len(shape) == 2:
+        if shape[1] != 3:
+            raise IncorrectShapeError(expected_shape="(n, 3)", actual_shape=str(shape))
     else:
-        raise ValueError('box_angles array with has not the correct shape.')
-
-    return output
-
-def digest_box_angles(box_angles):
-
-    output = None
-    unit = puw.get_unit(box_angles)
-    box_angles_value = puw.get_value(box_angles)
-    box_angles_value = digest_box_angles_value(box_angles_value)
-    output = box_angles_value*unit
-
-    return output
-
+        raise IncorrectShapeError(expected_shape="(n, 3)", actual_shape=str(shape))
+
+    return box_vectors
+
+
+def digest_box_vectors(box_vectors):
+    """ Checks if box vectors have the correct shape. Can be used
```

Comment:
The functions `digest_box_angles ` and `digest_box_lengths` were exactly equal, that's why I merged them into a single function, and I named it `digest_box_vectors` because I didn't know which name to use. However, we can split it into two functions in the next commit.

Reply:
I see your point. Then, let's approve these changes with a couple of things to be reviewed for milestone 0.6.1 (see https://github.com/uibcdf/MolSysMT/issues/59):
- The name of the function
- Whether or not makes sense to include digest_box_angles and digest_box_lengths temporary.